src/__tests__/storage.test.ts

```diff
@@ -0,0 +1,84 @@
+import { window } from '../../src/utils/globals'
+import { resetSessionStorageSupported, seekFirstNonPublicSubDomain, sessionStore } from '../storage'
+
+describe('sessionStore', () => {
+    describe('seekFirstNonPublicSubDomain', () => {
+        const mockDocumentDotCookie = {
+            value_: '',
+
+            get cookie() {
+                return this.value_
+            },
+
+            set cookie(value) {
+                //needs to refuse known public suffixes, like a browser would
+                // value arrives like dmn_chk_1699961248575=1;domain=.uk
+                const domain = value.split('domain=')
+                if (['.uk', '.com', '.au', '.com.au', '.co.uk'].includes(domain[1])) return
+                this.value_ += value + ';'
+            },
+        }
```

Comment:
@benjackwhite I couldn't get testcafe to test this. But I was having to attach the function to the window and then have testcafe eval it and who knows why it wasn't working.

Sometimes it wasn't present, others it returned the empty string, I think even sometimes it worked, but it was horrible.

So instead I have a fake cookie jar to test the logic with.

And then in the actual code...